WooCommerce/Classes/ViewRelated/Filters/FilterHistoryView.swift

```diff
@@ -0,0 +1,173 @@
+import SwiftUI
+
+/// Hosting controller for `FilterHistoryView`
+final class FilterHistoryViewHostingController<ViewModel: FilterListViewModel>: UIHostingController<FilterHistoryView<ViewModel>> {
+    init(viewModel: ViewModel, onSelection: @escaping (ViewModel.Criteria) -> Void) {
+        super.init(rootView: FilterHistoryView(viewModel: viewModel, onSelection: onSelection))
+        rootView.onDismiss = { [weak self] in
+            self?.dismiss(animated: true)
+        }
+    }
+
+    @MainActor @preconcurrency required dynamic init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+}
+
+/// View to list all saved filter history
+struct FilterHistoryView<ViewModel: FilterListViewModel>: View {
+
+    /// to be set externally on the hosting controller
+    var onDismiss: () -> Void = {}
+
+    private let viewModel: ViewModel
+    private let onSelection: (ViewModel.Criteria) -> Void
+
+    @State private var selectedFilter: ViewModel.Criteria?
+    @State private var savedFilters: [ViewModel.Criteria] = []
+    @State private var error: Error?
+    @State private var shouldConfirmClearingHistory = false
```

Comment:
It's not possible to have a generic observable object as a view model, so I'm keeping the states in the view for simplicity.